foundations/Cargo.toml

```diff
@@ -55,6 +55,9 @@ settings = [
     "dep:indexmap",
 ]
 
+# Adds #[serde(deny_unknown_fields)] to all settings containers using #[settings]
+settings-deny-unknown-fields = ["foundations-macros?/settings-deny-unknown-fields"]
```

Comment:
Let's make it a macro parameter instead, e.g. `#[settings(deny_unknown_fields = false)]`. With `true` being default in 4.0.0

Reply:
Done. My initial opposition to this approach was thinking that users of intermediate dependencies like Oxy wouldn't have a way to opt out for shared settings structs. But, in that case, that intermediate dependency can add its own feature flag and use it for the shared settings structs like so:
```
#[cfg_attr(
    feature = "settings-deny-unknown-fields",
    settings(deny_unknown_fields = false)
)]
#[cfg_attr(not(feature = "settings-deny-unknown-fields"), settings)]
```